### PR TITLE
Switch to exact time matching, remove matching by delta time

### LINKED ISSUE #31: Matching riders and drivers with time
Possible solution: rider selects all the times willing to go in the form, if none of the rider times match a driver's time, rider the rider doesn't get a driver.

### PR DESCRIPTION
**Pull request by rkalnins**
Wait to merge until #45 is merged. #45 may change matching process behavior.

Match riders with drivers using exact times instead of deltas. If a driver matches any of the rider's times exactly, the driver is considered compatible. The best driver is the driver leaving the earliest. We may need to discuss defining best as earliest departing. Currently, the reasoning is an earlier departure usually leads to more climbing time.

Close #31

### CODE REVIEW

File: scheduler/generate_rides.py [hunk 1]
```diff
@@ -101,16 +101,16 @@ def time_compatibility(rider: Rider, driver: Driver, day: Day.DayName) -> float:
...
     """
 
     rider_times = rider.get_times(day)
     driver_times = driver.get_times(day)
 
     rider_times.sort()
 
-    # there should only be one time here
+    # there should only be one time here, so maybe remove
+    # maybe keep this in case there are more than 1 to keep things deterministic
```
Comment (nzsilverman) [inline scheduler/generate_rides.py:114]: I agree, we should keep this to keep things deterministic... unless we add some error checking/ validation for the driver when we read their responses in. That would be a good thing to add, but for now this "hacks" around that just fine

File: scheduler/generate_rides.py [hunk 2]
```diff
@@ -101,16 +101,16 @@ def time_compatibility(rider: Rider, driver: Driver, day: Day.DayName) -> float:
             Day.DayName enum of day for which to check time compatability
 
     Returns:
-        Float that is the minimum time difference between when a specific driver and rider would
-        like to leave
+        Float of the earliest matching departure time in the rider and driver
```
Comment (nzsilverman) [inline scheduler/generate_rides.py:104]: Doesn't this also return -1 if no compatible time was found?

File: scheduler/generate_rides.py [hunk 3]
```diff
@@ -119,14 +119,16 @@ def time_compatibility(rider: Rider, driver: Driver, day: Day.DayName) -> float:
 
     driver_time = driver_times[0]
 
-    result = sys.maxsize
+    result = -1
```
Comment (nzsilverman) [inline scheduler/generate_rides.py]: This is why I commented above about -1

File: scheduler/tests/test_data.py [hunk 4]
```diff
@@ -193,7 +193,7 @@ class GenerateRidesTestData:
                 }]
             },
             "TUESDAY",
-            4,
+            -1,
```
Comment (nzsilverman) [inline scheduler/tests/test_data.py:196]: What is this -1 signifying here? I know this is test data, but I am curious what it is symbolizing

**Review by nzsilverman — CHANGES_REQUESTED** [review]
Just a few questions to be addressed, mainly about the return type of -1. That should be added in the comments, if it is intentional. Otherwise looks good.

File: scheduler/generate_rides.py (hunk 2 above)
Comment (rkalnins) [reply scheduler/generate_rides.py:104]: Yes, wan't sure if -1 would implicitly be converted to float. I think int to float conversion can be implicit. New commit makes change explicit. We return -1 if no match was found

File: scheduler/tests/test_data.py (hunk 4 above)
Comment (rkalnins) [reply scheduler/tests/test_data.py:196]: No match was found, see the comments above. Returns -1 if no match.

**Review by nzsilverman — APPROVED** [review]

**Pull request merged**